rpcclient/roundrobin.go

```diff
@@ -0,0 +1,367 @@
+// Copyright (c) 2015-2017 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package rpcclient
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/btcsuite/websocket"
+)
+
+var (
+	// ErrNoConnAvailable indicates no Conn is available for pick().
+	ErrNoConnAvailable = errors.New("no Connection is available")
+	// ErrNoUsableConnAvailable indicates no usable connections available i.e.
+	// client is marked as Shutdown.
+	ErrNoUsableConnAvailable = errors.New("no Connection is available for use, all are marked as Shutdown")
+)
+
+// ConnectionState indicates the current state of connection.
+type ConnectionState int
```

Comment:
None of this is used outside of the `rpcclient`, so it should all be unexported.

Reply:
Now since its used in balancer interface, we need to export it.